test/runtests.jl

```diff
@@ -16,7 +16,7 @@ end
     @test @inferred(mc(5.0)) == sin(5.0)
 
     # Default constructor.
-    mc_default = MistyClosure(OpaqueClosure(ir; do_compile=true), ir)
+    mc_default = MistyClosure(OpaqueClosure(ir; do_compile=true), Ref(ir))
```

Comment:
I wonder if should overload this so that
```suggestion
    mc_default = MistyClosure(OpaqueClosure(ir; do_compile=true), ir)
```

Reply:
I guess my feeling on this is that the recommended constructor (see a couple of lines above this test) is the one that I want people to use in general, and it does this, whereas this is the one that you use if the recommended constructor doesn't work as required for some reason, and so I'm less fussed about people having to put a bit more effort in to make it work for them.

Do you think this is acceptable, or should we add an additional constructor?

Reply:
reasonable

Reply:
Lovely. Thanks for the review -- I'm going to merge.